packages/dds/merge-tree/src/client.ts

```diff
@@ -1084,7 +1084,19 @@ export class Client {
      * @returns - segment and offset to slide the reference to
      */
     getSlideToSegment(segoff: { segment: ISegment | undefined; offset: number | undefined; }) {
-        return this.mergeTree._getSlideToSegment(segoff);
+        if (segoff.segment === undefined) {
+            return segoff;
+        }
+        const segment = this.mergeTree._getSlideToSegment(segoff.segment);
+        if (segment === segoff.segment) {
+            return segoff;
+        }
+        const offset =
```

Comment:
test failure is because previously we'd return -1 to indicate detachedness in the case `segment === undefined` here, but new logic returns 0. Not sure the baseline is particularly client-friendly, though we should probably save semantic changes of how clients deal with detached reference positions for a breaking change.